File: backend-go/internal/model/watchlist.go
package model

import "time"

// Watchlist 自选基金模型
type Watchlist struct {
	ID        string    `gorm:"type:uuid;primaryKey;default:uuid_generate_v4()" json:"id"`
	UserID    string    `gorm:"type:uuid;index" json:"user_id"`
	FundID    string    `gorm:"type:uuid;index" json:"fund_id"`
	Remark    *string   `gorm:"type:varchar(256)" json:"remark,omitempty"`
	SortOrder int       `gorm:"default:0" json:"sort_order"`
	CreatedAt time.Time `json:"created_at"`

	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Fund Fund `gorm:"foreignKey:FundID" json:"fund,omitempty"`
}

func (Watchlist) TableName() string { return "watchlist" }

File: ai-service-python/app/models/watchlist.py
from sqlalchemy import String, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, UUIDMixin


class Watchlist(UUIDMixin, Base):
    __tablename__ = "watchlist"

    user_id: Mapped[str] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    fund_id: Mapped[str] = mapped_column(UUID(as_uuid=True), ForeignKey("funds.id", ondelete="CASCADE"), nullable=False)
    remark: Mapped[str | None] = mapped_column(String(256))
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
